ui/pages/confirmations/hooks/transactions/useDappSwapComparisonInfo.ts

```diff
@@ -0,0 +1,324 @@
+    });
+    return result;
+  }, [chainId, tokenAddresses?.length]);
+
+  const getUSDValue = useCallback(
+    (tokenAmount: string, tokenAddress: Hex) => {
+      if (!tokenDetails || !fiatRates) {
+        return '0';
+      }
+      const decimals = new BigNumber(
+        Math.pow(
+          10,
+          parseInt(
+            getTokenValueFromRecord<TokenStandAndDetails>(
+              tokenDetails,
+              tokenAddress,
+            )?.decimals ?? '18',
+            10,
+          ),
+        ),
+      );
+      const conversionRate = new BigNumber(
+        getTokenValueFromRecord(fiatRates, tokenAddress) ?? 0,
+      );
+      return new BigNumber(tokenAmount ?? 0)
+        .dividedBy(decimals)
+        .times(conversionRate)
+        .toString(10);
+    },
+    [fiatRates, tokenDetails],
+  );
+
+  const getUSDValueForDestinationToken = useCallback(
+    (tokenAmount: string) => {
+      if (!quotesInput) {
+        return '0';
+      }
+      const { destTokenAddress } = quotesInput;
+      return getUSDValue(tokenAmount, destTokenAddress as Hex);
+    },
+    [fiatRates, getUSDValue, quotesInput],
+  );
+
+  const { value: quotes } = useAsyncResult<
+    QuoteResponse[] | undefined
+  >(async () => {
+    if (!quotesInput) {
+      return undefined;
+    }
+
+    captureDappSwapComparisonMetricsProperties({
+      properties: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        dapp_swap_comparison: 'loading',
+      },
+    });
+
+    const startTime = new Date().getTime();
+    updateQuoteRequestLatency();
+
+    const quotesList = await fetchQuotes(quotesInput);
+
+    updateQuoteResponseLatency(startTime);
+    return quotesList;
+  }, [
+    captureDappSwapComparisonMetricsProperties,
+    quotesInput,
+    requestDetectionLatency,
+  ]);
+
+  const getGasUSDValue = useCallback(
+    (gasValue: BigNumber) => {
+      if (!maxFeePerGas) {
+        return '0';
+      }
+      const gasPrice = new BigNumber(maxFeePerGas, 16);
+      const totalGas = gasPrice.times(gasValue).toString(10);
+      const nativeTokenAddress = getNativeTokenAddress(chainId);
+      return getUSDValue(totalGas, nativeTokenAddress);
+    },
+    [chainId, getUSDValue, maxFeePerGas],
+  );
+
+  useEffect(() => {
+    try {
+      if (
+        !amountMin ||
+        !tokenDetails ||
+        !fiatRates ||
+        !quotes?.length ||
+        !quotesInput ||
+        !simulationData
+      ) {
+        return;
+      }
+
+      const selectedQuote = getBestQuote(
+        quotes,
+        getUSDValueForDestinationToken,
+        getGasUSDValue,
+      );
+
+      if (!selectedQuote) {
+        return;
+      }
+
+      updateSwapComparisonLatency();
+
+      const { destTokenAddress, srcTokenAmount, srcTokenAddress } = quotesInput;
+      const {
+        approval,
+        quote: { destTokenAmount, minDestTokenAmount },
+        trade,
+      } = selectedQuote;
+
+      const totalGasInQuote = getGasUSDValue(
```

Comment:
Don't we already calculate this inside `getBestQuote`, could we just return it from `getBestQuote` via an object?